api/app/Http/Controllers/BaseController.php

```diff
@@ -1,39 +1,67 @@
 <?php namespace App\Http\Controllers;
 
+use App\Http\Transformers\BaseTransformer;
 use Laravel\Lumen\Routing\Controller;
 use Symfony\Component\HttpKernel\Exception\HttpExceptionInterface;
 use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Illuminate\Http\Request;
+// use Illuminate\Container\Container as App;
 
 class BaseController extends Controller
 {
-
     public static $model;
 
+    /**
+     * Transformer for the response arrays.
+     *
+     * @var App\Services\Transformer
+     */
+    protected $transformer;
+
+    /**
+     * Model Repository.
+     *
+     * @var App\Repositories\BaseRepository
+     */
+    protected $repository;
+
+    /**
+     * Validation service.
+     *
+     * @var App\Services\Validation\TestValidator
+     */
+    protected $validator;
+
+    /**
+     * Get all entities.
+     *
+     * @return Response
+     */
     public function getAll()
     {
-        $model = static::$model;
-
-        $entities = $model::all();
-
-        return response()->json($entities);
-
-//        $this->transformer = $entity.'Transformer';
-//        return $this->response->collection($entities, new $this->transformer);
+        return $this->transformer->collection($this->repository->all(), new BaseTransformer);
```

Comment:
@artstorm Yea I was leaning towards reducing the duplicated code in the concrete controllers, but I didn't realise that Laravel can't do DI in class hierarchies. I'm happy with your approach now that you've explained it.

I'd like to lean away from using Facades if possible as I have a feeling that the performance of facades is low, based on the fact that Lumen comes with Facades disabled by default. That may not be the case, but it seems likely as class discovery has to happen with a facade.